sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -335,7 +354,12 @@ class DataFrame private[sql](
    * @group action
    * @since 1.3.0
    */
-  def show(): Unit = show(20)
+  def show(): Unit = {
+    show(20)
+    if (this.count() > 20) {
```

Comment:
a better way to do this check is to implement it in showString. Instead of taking n, take n + 1, and if there are n+1 records, print  the line